src/main/java/net/neoforged/neoforge/transfer/EnergyHandlerUtil.java

```
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer;
+
+import net.minecraft.CrashReport;
+import net.minecraft.ReportedException;
+import net.minecraft.util.Mth;
+import net.minecraft.world.Container;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.redstone.Redstone;
+import net.neoforged.neoforge.transfer.handlers.energy.IEnergyHandler;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.Range;
+
+/**
+ * Utility class for handling various {@link IEnergyHandler} interactions
+ */
+public final class EnergyHandlerUtil {
+    /**
+     * @return True if the specified amount is 0 and should skip being processed.
+     * @throws ReportedException when amount is negative.
+     */
+    public static boolean checkEnergy(int amount) {
+        if (amount < 0) {
+            CrashReport report = CrashReport.forThrowable(new IllegalArgumentException("Amount must be non-negative"), "Energy amount was negative");
+            report.addCategory("EnergyHandlerUtil#isEmpty")
+                    .setDetail("Amount", amount);
+            throw new ReportedException(report);
```

If we are going to make this throw instead of just treat values less than zero as zero for short-circuiting purposes; then all the callers should be documented as being able to throw as well so that people know what gets thrown if an invalid value is passed.

---

I still need to go through and handle the documentations of throws in theory (as well as other documentations 🥹)